### PR TITLE
add timestamp to fail message

### PR DESCRIPTION
**Pull request by deeplow**
In order to improve the ease of finding the correct video segment and appropriate location for each test in the syslogs.

### CODE REVIEW

**Comment by marmarek** [comment]
The patch fails to apply (see https://openqa.qubes-os.org/tests/26824#step/system_tests/16 for example). nose2 version in fc32-based dom0 is different I guess. Plus, it needs to work also on fc25-based dom0 in R4.0...

**Comment by deeplow** [comment]
OK. I've now made two patches: one for 4.0 and one for 4.1+. Hopefully this works until it gets merged upstream and distributed downstream.

File: extra-files/convert_junit.py [hunk 1]
```diff
@@ -41,17 +41,27 @@ def main():
...
+                content_element = testcase.find('skipped')
                 testcase.set('status', 'skipped')
                 skipped += 1
             else:
                 testcase.set('status', 'success')
+
+            if content_element:
+                timestamp = testcase.get('timestamp')
+                content_element.text = "# timestamp {}\n{}".format(
+                        timestamp, content_element.text)
```
Comment (marmarek) [inline extra-files/convert_junit.py:63]: This doesn't seem to work. See https://openqa.qubes-os.org/tests/26998. `timestamp` is included in the XML, but not in the textual message.

Comment (deeplow) [reply extra-files/convert_junit.py]: Hum. I thought I had tested this. Should work now. Sorry about that.

Comment (marmarek) [reply extra-files/convert_junit.py]: I'm having troubles to see how the new version is different than the old one (besides moving to a function)? Which part makes it work now, order of operations?

**Comment by marmarek** [comment]
There is a case of failing nose2: https://openqa.qubes-os.org/tests/27045#step/system_tests/36
```
ERROR:nose2.main:Internal Error
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/nose2/main.py", line 266, in runTests
    self.result = runner.run(self.test)
  File "/usr/lib/python3.8/site-packages/nose2/runner.py", line 53, in run
    executor(test, result)
  File "/usr/lib/python3.8/site-packages/nose2/runner.py", line 41, in <lambda>
    executor = lambda suite, result: suite(result)
  File "/usr/lib64/python3.8/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib64/python3.8/unittest/suite.py", line 122, in run
    test(result)
  File "/usr/lib64/python3.8/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib64/python3.8/unittest/suite.py", line 122, in run
    test(result)
  File "/usr/lib64/python3.8/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib64/python3.8/unittest/suite.py", line 122, in run
    test(result)
  File "/usr/lib64/python3.8/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib64/python3.8/unittest/suite.py", line 122, in run
    test(result)
  File "/usr/lib64/python3.8/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib64/python3.8/unittest/suite.py", line 122, in run
    test(result)
  File "/usr/lib64/python3.8/unittest/case.py", line 736, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib64/python3.8/unittest/case.py", line 684, in run
    self._feedErrorsToResult(result, outcome.errors)
  File "/usr/lib64/python3.8/unittest/case.py", line 600, in _feedErrorsToResult
    result.addFailure(test, exc_info)
  File "/usr/lib/python3.8/site-packages/nose2/result.py", line 69, in addFailure
    self.session.hooks.testOutcome(event)
  File "/usr/lib/python3.8/site-packages/nose2/events.py", line 230, in __call__
    result = getattr(plugin, self.method)(event)
  File "/usr/lib/python3.8/site-packages/nose2/plugins/junitxml.py", line 134, in testOutcome
    str(self._timestamp_to_iso8601(self._start)))
  File "/usr/lib/python3.8/site-packages/nose2/plugins/junitxml.py", line 253, in _timestamp_to_iso8601
    return datetime.datetime.utcfromtimestamp(timestamp).isoformat()
TypeError: an integer is required (got type NoneType)
Internal Error: runTests aborted: an integer is required (got type NoneType)
```

**Comment by deeplow** [comment]
> There is a case of failing nose2: https://openqa.qubes-os.org/tests/27045#step/system_tests/36

I tried reproducing this locally but couldn't get the same. So I don't know exactly what conditions lead to this case. Anyhow, I noticed the other time-related function in the nose2 junitxml plugin already handled this exception by setting time to 0. So I did the same for the timestamp. Not sure if the best approach but didn't think of another one.

So in these cases instead of it failing now, it should instead set the timestamp of 0 (`'1970-01-01T00:00:00'`). Also added this change to the PR upstream to see what the nose devs say.

**Comment by deeplow** [comment]
> There is a case of failing nose2: https://openqa.qubes-os.org/tests/27045#step/system_tests/36
> 
> ```
> ERROR:nose2.main:Internal Error
> Traceback (most recent call last):
>   File "/usr/lib/python3.8/site-packages/nose2/main.py", line 266, in runTests
>     self.result = runner.run(self.test)
>   File "/usr/lib/python3.8/site-packages/nose2/runner.py", line 53, in run
>     executor(test, result)
>   File "/usr/lib/python3.8/site-packages/nose2/runner.py", line 41, in <lambda>
>     executor = lambda suite, result: suite(result)
>   File "/usr/lib64/python3.8/unittest/suite.py", line 84, in __call__
>     return self.run(*args, **kwds)
>   File "/usr/lib64/python3.8/unittest/suite.py", line 122, in run
>     test(result)
>   File "/usr/lib64/python3.8/unittest/suite.py", line 84, in __call__
>     return self.run(*args, **kwds)
>   File "/usr/lib64/python3.8/unittest/suite.py", line 122, in run
>     test(result)
>   File "/usr/lib64/python3.8/unittest/suite.py", line 84, in __call__
>     return self.run(*args, **kwds)
>   File "/usr/lib64/python3.8/unittest/suite.py", line 122, in run
>     test(result)
>   File "/usr/lib64/python3.8/unittest/suite.py", line 84, in __call__
>     return self.run(*args, **kwds)
>   File "/usr/lib64/python3.8/unittest/suite.py", line 122, in run
>     test(result)
>   File "/usr/lib64/python3.8/unittest/suite.py", line 84, in __call__
>     return self.run(*args, **kwds)
>   File "/usr/lib64/python3.8/unittest/suite.py", line 122, in run
>     test(result)
>   File "/usr/lib64/python3.8/unittest/case.py", line 736, in __call__
>     return self.run(*args, **kwds)
>   File "/usr/lib64/python3.8/unittest/case.py", line 684, in run
>     self._feedErrorsToResult(result, outcome.errors)
>   File "/usr/lib64/python3.8/unittest/case.py", line 600, in _feedErrorsToResult
>     result.addFailure(test, exc_info)
>   File "/usr/lib/python3.8/site-packages/nose2/result.py", line 69, in addFailure
>     self.session.hooks.testOutcome(event)
>   File "/usr/lib/python3.8/site-packages/nose2/events.py", line 230, in __call__
>     result = getattr(plugin, self.method)(event)
>   File "/usr/lib/python3.8/site-packages/nose2/plugins/junitxml.py", line 134, in testOutcome
>     str(self._timestamp_to_iso8601(self._start)))
>   File "/usr/lib/python3.8/site-packages/nose2/plugins/junitxml.py", line 253, in _timestamp_to_iso8601
>     return datetime.datetime.utcfromtimestamp(timestamp).isoformat()
> TypeError: an integer is required (got type NoneType)
> Internal Error: runTests aborted: an integer is required (got type NoneType)
> ```

Also @marmarek, where do I find these logs in text form? I grepped in all the logs and assets and couldn't find it.

**Comment by marmarek** [comment]
> Also @marmarek, where do I find these logs in text form? I grepped in all the logs and assets and couldn't find it.

In the `system_tests-tests-*.log` file.

**Review by marmarek — APPROVED** [review]

File: tests/system_tests.pm [hunk 2]
```diff
@@ -33,6 +33,11 @@ sub run {
     if (check_var("VERSION", "4.0")) {
         # until https://github.com/nose-devs/nose2/pull/412 gets merged
         assert_script_run("sudo patch /usr/lib/python3*/site-packages/nose2/plugins/junitxml.py /root/extra-files/nose2-junit-xml-log-skip-reason.patch");
+        # until https://github.com/nose-devs/nose2/pull/505 gets merged
+        assert_script_run("sudo patch /usr/lib/python3*/site-packages/nose2/plugins/junitxml.py /root/extra-files/nose2-junit-xml-timestamp.r4.0.patch");
+    } else {
+        # until https://github.com/nose-devs/nose2/pull/505 gets merged
+        assert_script_run("sudo patch /usr/lib/python3*/site-packages/nose2/plugins/junitxml.py /root/extra-files/nose2-junit-xml-timestamp.r4.1.patch");
```
Comment (deeplow) [inline tests/system_tests.pm:40]: @marmarek this was now merged upstream. Soon it should be in a release :)

**Pull request merged**